src/structured-source.ts

```diff
@@ -24,33 +24,31 @@
 
 import { upperBound } from 'boundary';
 
-export class Position {
-    constructor(line, column) {
-        this.line = line;
-        this.column = column;
-    }
+export type SourceRange = readonly [number, number];
+export type SourcePosition = {
+    readonly line: number,
+    readonly column: number,
 }
 
-export class SourceLocation {
-    constructor(start, end) {
-        this.start = start;
-        this.end = end;
-    }
+export type SourceLocation = {
+    readonly end: SourcePosition,
+    readonly start: SourcePosition,
 }
```

Comment:
Previously, `SourceLocation` was Class.
However, I read tests and these test treat `SourceLocation` as JSON.

I think that we can represent `SourceLocation` as types.
So, I change `SourceLocation` class to `SourceLocation` type.

Reply:
Previous implementaion fail next test.

```js
const src = new StructuredSource('');
assert.deepStrictEqual(src.locationToRange({ // strict equal does not equal JSON and Class
    start: { line: 1, column: 0 },
    end: { line: 1, column: 2 }
}), [0, 2]);
````